meetings/working-groups/collection-literals/meetings/working-groups/collection-literals/CL-2023-06-16.md

```diff
@@ -0,0 +1,125 @@
+
+THe compelling case to want to make this a cast of a collection literal is:
+
+```c#
+using CrazyArray = ImmutableArray<List<HashSet<int>>; // i.e. some complex collection type
+
+// ...
+var v1 = (CrazyArray)[a, b, c]; // really don't want to have to spell out the type in each location
+var v2 = (CrazyArray)[x, y, z];
+```
+
+Currently the prototype impl makes no breaks here and keeps parsing as it exists prior to this feature.
+
+As this is a breaking change, we might want to combine it with the work on "safe breaking changes" being looked into.  However, we also think the chance of this impacting someone would be super small (though investigations would be necessary).  We want to get a read from LDM on if we should explore this fruther.
+
+### Extension method lookup
+
+WG believes it would be valuble for customers to be able to write:
+
+```c#
+static class Extensions
+{
+    public static ImmutableArray<T> AsImmutableArray<T>(this ImmutableArray<T> array) => array;
+}
+
+void M()
+{
+    var v = [complex, type, examples].AsImmutableArray();
+}
+```
+
+This is because the alternative would be to have to write something like:
+
+```c#
+(ImmutableArray<Complex<Type<Examples[]>>)[complex, type, examples];
+```
+
+In other words, the extension method approach would allow for type inference to make this much simpler and cleaner than having to fully specify all the types.  Working group believes (but needs to verify) that a simple extension (no pun intended) to 'extension method lookup' would make this work.  Specifically, while the collection type has no natural type currently (like `null`, a `lambda` or a `method group`) it could still trivially participate in extension method lookup and would work in cases like this.
+
+We do, however, believe the value here is diminished if, say, the language adopts features like "postfix cast" and "partial generic type inference".  If those features make progress, you could write the above as:  ```[complex, type, examples].(ImmutableArray<_>)```.  This would be simple and clear.  
+
+WG also thinks that if we did add extension method support for the above, that we could consider doing the same for lambdas and method-groups.
+
+### Syntactic diambiguation update
+
+The feature has some small syntactic ambiguities with both existing features, and features coming around at the same time.  Specifically:
+
+Case 1:
+```c#
+a ? [b] ...   // is that:
+
+(a?[b])       // a conditional access expression, or:
+a ? [b] : ... // a conditional expression
+```
+
+Conclusion: we will preserve existing parsing for these.  So if it is a case where it would successfully parse as a conditional-access-expression, it will remain that way.  *However*, if trying that approach would end up failing, we will try again as a conditional-expression and accept that if it succeeds.  This means the following work:
```

Comment:
?
```suggestion
Conclusion: we will preserve existing parsing for these.  So if it is a case where it would successfully parse as a conditional-access-expression, it will remain that way.  *However*, if trying that approach would end up failing, we will try again as a conditional-expression and accept that if it succeeds.  This means the following will work:
```